linode/manager
# feat: [M3-6973] - Add DC-specific pricing Object Storage overages

## Description 📝
Object storage has hardcoded prices throughout the codebase. The service is a flat $5/month. Beyond the first 250GB storage and 1TB outbound transfer, it's and $0.02/GB/month. We need to update this due to pricing changes for DC-specific overage rates and clarification that outbound transfer is *not* added to DC-specific pools, only the global pool.

The only time a user sees the Enable Object Storage notice is the first time they create a bucket or access key and must opt-in to the service. 
- For this reason, we are adding region-specific overage pricing to the Create Bucket drawer.

Note: Access keys are not currently tied to regions.

## Changes  🔄
- Overage prices on the Create Bucket drawer reflect accurate prices for dynamic region pricing.
- Copy updates are made in the Enable Object Storage modal to match the mocks for enabling via both bucket and key.
- Cleaned up `dcSpecificPricing` feature flag in the Enable OBJ Modal and related tests, since initial DC-specific pricing work has been released, to avoid extra conditional logic. DC-specific pricing behavior is now the default.
- Updated tests and added a new unit test for a new component.

## Preview 📷
| Description | Prod | This Branch |
| ---- | ---- | ---- |
| Enable Object Modal | ![Screenshot 2023-10-18 at 11 26 56 AM](https://github.com/linode/manager/assets/114685994/297cb31e-0412-4f02-a07c-a22007dd292d) ![Screenshot 2023-10-18 at 11 16 30 AM](https://github.com/linode/manager/assets/114685994/acefa512-98bd-465d-af51-a73621a268e5) ![Screenshot 2023-10-18 at 11 17 49 AM](https://github.com/linode/manager/assets/114685994/b47be026-ae63-4165-9c2e-2a90b5685bc5) | ![Screenshot 2023-10-18 at 11 16 54 AM](https://github.com/linode/manager/assets/114685994/483f7362-5cde-4e1f-94d2-9e5a7aec14a5) ![Screenshot 2023-10-18 at 11 17 33 AM](https://github.com/linode/manager/assets/114685994/905097c9-b1d4-4a2c-b161-6ac5d6a6ad9e) |
| Create Bucket Drawer | ![Screenshot 2023-10-18 at 11 14 25 AM](https://github.com/linode/manager/assets/114685994/1f09d949-1322-41fb-aad5-069122935eb5) | ![Screenshot 2023-10-18 at 11 02 18 AM](https://github.com/linode/manager/assets/114685994/25f983a6-1fce-440f-b694-600b8dd2059d) ![Screenshot 2023-10-18 at 11 02 26 AM](https://github.com/linode/manager/assets/114685994/454fc522-836f-4c18-b90b-8d053cb5654d)  ![Screenshot 2023-10-18 at 11 02 34 AM](https://github.com/linode/manager/assets/114685994/ad6ac4bd-53b3-4d7f-9116-68ec0d16cd95)|
| Create Bucket Drawer Tooltip | N/A | ![Screenshot 2023-10-18 at 11 02 44 AM](https://github.com/linode/manager/assets/114685994/80403bcc-0b41-4a3d-93c4-8ce8305519c0) ![Screenshot 2023-10-18 at 11 02 56 AM](https://github.com/linode/manager/assets/114685994/cb574db4-eb76-4498-84ff-0bfd1a3b59c6) ![Screenshot 2023-10-18 at 1 07 51 PM](https://github.com/linode/manager/assets/114685994/4d3f224e-6a99-4444-9ddd-61fb976d4fe0)|

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Check out this PR.
- Turn on the `objDCSpecificPricing` feature flag.
- If Object Storage already enabled, go to /account/settings and cancel it.

### Verification steps 
- With the feature flags on:
   - Confirm that the copy of the modal matches the copy in the screenshots above for each scenario, which matches the mocks attached to the internal ticket.
   - [ ] Create your first access key and enable OBJ.
   - [ ] Create your first bucket and enable OBJ.
   - Confirm that overage pricing displays in the Create Bucket drawer, matches the mocks attached to the internal ticket, and updates dynamically depending on cluster region selection.
   - [ ] Create Bucket in Sao Paulo.
   - [ ] Create Bucket in Jakarta.
   - [ ] Create Bucket in any other region (will use base pricing).

Verify unit tests pass:
```
yarn test EnableObjectStorageModal OveragePricing CreateBucketDrawer
```
Verify Cypress test passes:
```
yarn cy:run -s "cypress/e2e/core/objectStorage/enable-object-storage.spec.ts"
```
   
## As an Author I have considered 🤔

*Check all that apply*
